File: java_solutions_11-20/Euler_20.java
/*
/   Project Euler problem #20 Factorial digit sum
/
/   Find the sum of the digits in the number 100!
/
*/

import java.math.BigInteger;


public class Euler_20
{
    public static void main(String[] args)
    {
        BigInteger oneHundredFactorial = new BigInteger("2");
        BigInteger nextFactorial;

        int sumOfDigits = 0;

        for (int i = 3; i <= 100; i++) {
            nextFactorial = new BigInteger(String.valueOf(i));
            oneHundredFactorial = oneHundredFactorial.multiply(nextFactorial);
        }

        String factorialString = oneHundredFactorial.toString();
        String[] factorialStringSplit = factorialString.split("");

        for (int i = 0; i < factorialStringSplit.length; i++) 
            if (!factorialStringSplit[i].equals("")) 
                sumOfDigits += Integer.parseInt(factorialStringSplit[i]);

        System.out.printf("Answer: %d\n", sumOfDigits);
    }

}

File: Python/py_solutions_11-20/Euler_20.py
# find the sum of the digits in !100 ? 

from __future__ import print_function

import math
import timeit


start = timeit.default_timer()

def euler_20():
    return sum(map(int, str(math.factorial(100))))

print("Answer: {}".format(euler_20()))
stop = timeit.default_timer()
print("Time: {0:9.5f}".format(stop - start))

